### PR TITLE
Update Pin icons & add Pin Unfilled icons

### LINKED ISSUE #6844: [Icon]: Update Pin Icon
## Why are you proposing this update?

With the introduction of a new Pin icon as part of the [App Navigation project](https://vault.shopify.io/projects/18380), there's now inconsistency with two different pin icons used throughout the admin. The new pin icon being proposed will help create consistency across the admin, and also remove the need for a separate "unpinned" icon that I believe is currently not part of Polaris (more on that below). 

app-icon-comparison

## Icon name

Pin

## What type of icon set?

Both Major and Minor

## Icon SVG code

Pin Major SVG:
PinMajor

Pin Minor SVG:
PinMinor

## Are there any other important details?

- [Link to Figma file](https://www.figma.com/file/1lhdbPTTEXGm2LyatXQAQE/App-Launcher-%5BWorking-file%5D?node-id=5110%3A121811)
- The current Pin icon in Polaris is a "solid" style, while the new icon being proposed is an "outline" style. Typically the Pin icon is used in two different states (pinned and unpinned), so the proposed outline style will provide developers the flexibility to modify a few CSS properties to change the icon state without needing to swap out the entire icon. If you want to see this pinned/unpinned state change in action, you can [pin an app to your navigation](https://user-images.githubusercontent.com/95929346/182708131-f99973ed-14fb-40b5-a729-2851992a199f.png).

### PR DESCRIPTION
**Pull request by chrisblinstrub**
### WHY are these changes introduced?

Fixes #6844 

### WHAT is this pull request doing?
This PR is updating the Polaris Pin icon (both Major & Minor versions) to be consistent throughout the Admin. It is also introducing a new "Pin Unfilled" icon to Polaris.

Note: There is currently a dependency on the [PinIcon component](https://github.com/Shopify/web/tree/main/app/components/PinIcon), which is being updated [here](https://github.com/Shopify/core-workflows/issues/525).

app-icon-comparison

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by nickpresta** [comment]
@chrisblinstrub Thanks for adding this to Polaris! We had planned on doing this after the Reports Index beta period wraps up, so we appreciate you proactively doing it.

**Comment by bowen9284** [comment]
Hi @chrisblinstrub any movement on getting these icons merged? We're currently implementing the Reports Index feature on mobile and would benefit from these new icons.

**Comment by chrisblinstrub** [comment]
Hey @bowen9284 - there were some dependencies with this PR that required us to first update the [`PinIcon` component](https://github.com/Shopify/web/tree/main/app/components/PinIcon), more details on those dependencies [here](https://github.com/Shopify/core-workflows/issues/525). 

@kragotte has added this to Optimize's board for the Reports index icon ([here](https://github.com/Shopify/core-issues/issues/47737)) so we can get these updates finalized.

File: .changeset/big-pandas-poke.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-icons': major
```
Comment (BPScott) [inline .changeset/big-pandas-poke.md]: This is a minor release, not a major.

**Comment by Rmnlly** [comment]
@chrisblinstrub Having pulled the unfilledicon from here into web (for removing the polaris hacks) we found that the `PinIconUnfilled` icon had a `fill="#5C5F62"` property. This is preventing us from filling it with css, so we should remove this property from all the icons as they'll have fills set in polaris/web.

**Comment by kyledurand** [comment]
From the failing test:
> packages/icons/PinMajor.svg › only has <path>s, <polygon>s and <circle>s with an explict fill color

Make sure any fills or strokes are using `#5C5F62`

**Review by sam-b-rose — APPROVED** [review]
Looks good to me 👍 📌  Only suggestion/thought, the SVG path data looks a little verbose. Should we run this through https://svgomg.net/ ? When doing this for one of the icons, I see a nice little reduction of ~35% in size.

Not a blocker imo! Just a suggested optimization

**Pull request merged**